pkg/deploy/api/v1beta3/types.go

```diff
@@ -254,6 +254,11 @@ type DeploymentConfigSpec struct {
 	// Replicas is the number of desired replicas.
 	Replicas int `json:"replicas" description:"the desired number of replicas"`
 
+	// Test ensures that this deployment config will have zero replicas except while a deployment is running. This allows the
+	// deployment config to be used as a continuous deployment test - triggering on images, running the deployment, and then succeeding
+	// or failing. Post strategy hooks and After actions can be used to integrate successful deployment with an action.
+	Test bool `json:"test" description:"if true, this deployment config will always be scaled to 0 except while a deployment is running"`
```

Comment:
Also, I think upstream is wrong about omitempty - because it overlaps with serialization rules it shouldn't be used for swagger like this.